### PR TITLE
vlc: add 3.0.4 recipe

### PR DESCRIPTION
**Pull request by kenmays**
Tested recipe on hrev52358 x86_64.

> **libvlc configuration**
> version               : 3.0.4
> system                : haiku
> architecture          : x86_64 mmx sse sse2
> optimizations         : yes
> vlc aliases           : cvlc rvlc qvlc nvlc

TAG: Work In Progress/help wanted/feedback needed (you can merge, but it needs patching at this time so leave deactvated))

### CODE REVIEW

File: media-video/vlc/vlc-3.0.4.recipe [hunk 1]
```diff
@@ -91,6 +95,8 @@ REQUIRES="
 	lib:libx265$secondaryArchSuffix
 	lib:libxml2$secondaryArchSuffix
 	lib:libz$secondaryArchSuffix
+	lib:libX11$secondaryArchSuffix
```
Comment (extrowerk) [inline media-video/vlc/vlc-3.0.4.recipe]: Is it mandatory? If not, pls remove.

File: media-video/vlc/vlc-3.0.4.recipe [hunk 2]
```diff
@@ -91,6 +95,8 @@ REQUIRES="
...
+	lib:libxcb$secondaryArchSuffix
```
Comment (extrowerk) [inline media-video/vlc/vlc-3.0.4.recipe]: And this?

File: media-video/vlc/vlc-3.0.4.recipe [hunk 3]
```diff
@@ -165,10 +170,11 @@ BUILD_REQUIRES="
 	devel:libvpx$secondaryArchSuffix
 	devel:libx264$secondaryArchSuffix
 	devel:libx265$secondaryArchSuffix
-	devel:libxcb$secondaryArchSuffix
 	devel:libxml2$secondaryArchSuffix
 	devel:libz$secondaryArchSuffix
 	devel:live555$secondaryArchSuffix
+#	devel:libX11$secondaryArchSuffix
```
Comment (extrowerk) [inline media-video/vlc/vlc-3.0.4.recipe]: Do not add X11 deps if they arent required.

File: media-video/vlc/vlc-3.0.4.recipe [hunk 4]
```diff
@@ -154,8 +152,8 @@ BUILD_REQUIRES="
 	devel:libQt5Widgets$secondaryArchSuffix >= 5.7
 	devel:libsamplerate$secondaryArchSuffix
 #	devel:libschroedinger_1.0$secondaryArchSuffix # fix libschroedinger.pc
-	devel:libsdl_1.2$secondaryArchSuffix
-	devel:libsdl_image_1.2$secondaryArchSuffix
+	devel:libSDL2_2.0$secondaryArchSuffix
+	devel:libSDL2_image_2.0$secondaryArchSuffix
```
Comment (diversys) [inline media-video/vlc/vlc-3.0.4.recipe]: Why SDL libs are needed? I thought VLC dropped SDL support in 3.x

**Comment by diversys** [comment]
Without an actual patch the recipe is useless.

**Comment by extrowerk** [comment]
Yep, i made an extensive research ( but not checked the actual sources yet, as i am on the road right now), but afaik they dropped the sdl support.

-- Z.

2018.09.25. dátummal, 11:41 időpontban Sergei Reznikov <notifications@github.com> írta:

**Comment by kenmays** [comment]
@diversys - The older VLC recipes are 'broken - this VLC recipe works for build/patch (for now). SDL isn't used, I'm still reviewing dependency issues.

File: media-video/vlc/vlc-3.0.4.recipe (hunk 4 above)
Comment (kenmays) [reply media-video/vlc/vlc-3.0.4.recipe]: Removed.

**Comment by extrowerk** [comment]
What kind of output could we use as alternative for SDL/SDL2?

-- Z.

2018.09.25. dátummal, 17:52 időpontban kenmays <notifications@github.com> írta:

**Comment by scottmc** [comment]
@kenmays Can you post a screenshot of it running?

**Comment by kenmays** [comment]
@scottmc - I will, once hurdles are passed.

**Comment by extrowerk** [comment]
Maybe you should explain with what kind of hurdles are you fighting to get some help. Just my 1 cents.

Oh, and 2 more things:
- Your commit message is always the same. Be more specific what did you changed and why. Provide info so one doesn't have to ask.

- If you make a PR for a known broken recipe, at least tell it in the description, or if you have plans to adjust/extend it then add a Do not merge flag, because one can mistakenly merge it, as the HaikuPorts team have almost no human resources for testing PRs and will expect working state (except fbrosson and alaviss, but they are advanced AI-s or superhumans or aliens, i think.) so be explicit to help others, and do not expect somebody test Your PR before the merge.

-- Z.

2018.09.25. dátummal, 21:20 időpontban kenmays <notifications@github.com> írta:

**Comment by kenmays** [comment]
@scottmc @extrowerk - The VLC 3.0.4 recipe isn't broken. We have xcb/X11 libs in Haiku Depot but I guess they are not for true usage? Anyhow, I built vlc-2.2.8 as seen here:
vlc-2 2 8

**Comment by kenmays** [comment]
We can close this PR. I feel that since we already have a nice supported version of SMPlayer and MediaPlayer, this is not needed for most casual users at this time.

**Comment by scottmc** [comment]
Please open a new issue if there isn't one already for vlc and attach this screenshot along with a summary of the known issues.  VLC is one we've kicked around for years.

**Pull request closed without merging**